apps/bfd-model/bfd-model-dsl-codegen/bfd-model-dsl-codegen-library/src/main/java/gov/cms/model/dsl/codegen/library/DataTransformer.java

```diff
@@ -0,0 +1,775 @@
+        addError(fieldName, "invalid integer");
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Copies an optional field only if its value exists. Uses lambda expressions for the existence
+   * test as well as the value extraction. Optional fields must be nullable at the database level
+   * but must return non-null values when the supplier is called.
+   *
+   * <p>Parses the string into a Integer. Valid null values are silently accepted without calling
+   * the Consumer.
+   *
+   * @param fieldName name of the field from which the value originates
+   * @param exists returns true if the value exists
+   * @param value returns the value to copy
+   * @param copier Consumer to receive the date
+   * @return this
+   */
+  public DataTransformer copyOptionalIntString(
+      String fieldName, BooleanSupplier exists, Supplier<String> value, Consumer<Integer> copier) {
+    if (exists.getAsBoolean()) {
+      return copyIntString(fieldName, false, value.get(), copier);
+    }
+    return this;
+  }
+
+  /**
+   * Parses the string into an Long and delivers it to the Consumer. The string value must be a
+   * valid long. Valid null values are silently accepted without calling the Consumer.
+   *
+   * @param fieldName name of the field from which the value originates
+   * @param nullable true if null is a valid value
+   * @param value long string
+   * @param copier Consumer to receive the date
+   * @return this
+   */
+  public DataTransformer copyLongString(
+      String fieldName, boolean nullable, String value, Consumer<Long> copier) {
+    if (nonNull(fieldName, value, nullable)) {
+      try {
+        long longValue = Long.parseLong(value);
+        copier.accept(longValue);
+      } catch (NumberFormatException ex) {
+        addError(fieldName, "invalid long");
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Copies an optional field only if its value exists. Uses lambda expressions for the existence
+   * test as well as the value extraction. Optional fields must be nullable at the database level
+   * but must return non-null values when the supplier is called.
+   *
+   * <p>Parses the string into a Long. Valid null values are silently accepted without calling the
+   * Consumer.
+   *
+   * @param fieldName name of the field from which the value originates
+   * @param exists returns true if the value exists
+   * @param value returns the value to copy
+   * @param copier Consumer to receive the date
+   * @return this
+   */
+  public DataTransformer copyOptionalLongString(
+      String fieldName, BooleanSupplier exists, Supplier<String> value, Consumer<Long> copier) {
+    if (exists.getAsBoolean()) {
+      return copyLongString(fieldName, false, value.get(), copier);
+    }
+    return this;
+  }
+
+  /**
+   * Parses the string into a LocalTimestamp and delivers it to the Consumer. The string value must
+   * be in ISO-8601 format (YYYY-MM-DD). Valid null values are silently accepted without calling the
```

Comment:
Is this function used to convert a date, a datetime value or either?  If there is a time element that is being parsed, do we know that the values are always in UTC?

Reply:
I'm wondering if we should include "UTC" in the function somewhere so its implied that this is always converting time values to that timezone

Reply:
Good question! The parsing code in this method is identical to that used in the current `RifParsingUtils#parseTimestamp` method so the use of UTC should be correct.  However the comment was misleading so I have changed it to reflect what is actually being done.